local.go

```diff
@@ -0,0 +1,549 @@
+package imgutil
+
+import (
+	"archive/tar"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+)
+
+type LocalImage struct {
+	RepoName         string
+	Docker           *client.Client
+	Inspect          types.ImageInspect
+	layerPaths       []string
+	Stdout           io.Writer
+	currentTempImage string
+	prevDir          string
+	prevMap          map[string]string
+	prevOnce         *sync.Once
+	easyAddLayers    []string
+}
+
+func EmptyLocalImage(repoName string, dockerClient *client.Client) *LocalImage {
```

Comment:
Do we need to inject the docker client? Can we just instantiate it in the constructor?

Reply:
This actually seems nice to me. What if you want to instantiate the client with different parameters (e.g., with different daemon location)? Given that this is a library, probably better to avoid reading Docker env vars directly.

Reply:
yeah that makes sense